### PR TITLE
Feat: Enabled Flag & Consistent Tags

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* `enabled` flag for individual SLOs and Synthetics
* Consistent Tag Methodology - A map is passed in and turned into an array via the module (which is what DD provider needs)

## why
* new features and consistency.
  * Tagging consistency fixes bug where tags show up `0:key:value` when wrong type is used.

### CODE REVIEW

File: modules/slo/main.tf [hunk 1]
```diff
@@ -47,7 +47,17 @@ resource "datadog_service_level_objective" "monitor_slo" {
...
+  # if a key is supplied with a value, it will render "key:value" as a tag
+  #   tags:
+  #     key = value
+  # if a key is supplied without a value (null), it will render "key" as a tag
+  #   tags:
+  #     key = null
+  tags = [
+    for tagk, tagv in lookup(each.value, "tags", module.this.tags) :
+    tagv != null ? format("%s:%s", tagk, tagv) : tagk
+  ]
```
Comment (nitrocode) [inline modules/slo/main.tf:60]: can this be moved to a local and copied for `datadog_service_level_objective.monitor_slo` and `datadog_service_level_objective.metric_slo`

File: modules/slo/variables.tf [hunk 2]
```diff
@@ -1,26 +1,5 @@
...
-    monitor_ids = list(number)
-
-    message      = string
-    description  = string
-    force_delete = bool
-    validate     = bool
-
-    tags = list(string)
-  }))
+  type        = any
```
Comment (goruha) [inline modules/slo/variables.tf:2]: Is structure too compicated to define it as type?

**Review by goruha — CHANGES_REQUESTED** [review]

File: modules/slo/variables.tf (hunk 2 above)
Comment (Benbentwo) [reply modules/slo/variables.tf:2]: The problem is not the structure, its that we want optional variables. we want enabled to be optional with a default of true. but if we add it to the struct then you have to explicitly say on every SLO that `enabled: true` which becomes a pain

**Review by nitrocode — APPROVED** [review]
Approved provided this is tested in a client with enabled flag on slos and synthetics also tested

**Pull request merged**